### DISCUSSION TITLE
i18n without modifying URL

### DISCUSSION THREAD

**Discussion by litari** [vercel/next.js#21852] [Help]
We're looking to migrate to next.js, but there is one big questionmark remaining.

We need internationalization-support **without** changing the url. When reading the docs and searching the web, the alternatives seem to be:

i18n with subpath-routing
i18n with domain-routing
custom server

what we want:
```
en: oursite.com/path
fr: oursite.com/path
```

what we dont want:
```
en: oursite.com/path
fr: oursite.fr/path or oursite.com/fr/path or fr.oursite.com/path
```

Hope I'm clear enough. Any suggestions?

**Comment by JustFly1984** [comment]
You can’t have what you want with SSR. The reason is SEO. Basically you can’t have meta tags for 2 languages on the same page. 

In case you do not care for SEO, you can disable SSR, and use CSR instead, detect language at runtime and render html after language detected.

**Reply by fpmanuel** [reply]
Thanks @JustFly1984! Clear enough :)
